File: app/src/main/java/ru/willdes/nginxplus/UpstreamName.java
package ru.willdes.nginxplus;

public class UpstreamName {
    private static UpstreamName upstreamName;
    private int idupstr;
    private String upstrname;

    public static UpstreamName getUpstreamName() {
        if (upstreamName == null) {
            upstreamName = new UpstreamName();
        }
        return upstreamName;
    }

    public int getIdupstr() {
        return idupstr;
    }

    public void setIdupstr(int idupstr) {
        this.idupstr = idupstr;
    }

    public String getUpstrname() {
        return upstrname;
    }

    public void setUpstrname(String upstrname) {
        this.upstrname = upstrname;
    }
}

File: app/src/main/java/ru/willdes/nginxplus/UpstreamName.kt
package ru.willdes.nginxplus

class UpstreamName {
    var idupstr = 0
    var upstrname: String? = null

    companion object {
        var upstreamName: UpstreamName? = null
            get() {
                if (field == null) {
                    field = UpstreamName()
                }
                return field
            }
            private set
    }
}